### ISSUE TITLE
Java monitoring

### ISSUE THREAD

**Issue by BGmot** [zabbix-community/helm-zabbix#23]
**Is your feature request related to a problem? Please describe.**
With this helm chart it is impossible to monitor Java applications via JMX.

**Describe the solution you'd like**
For some reason you forward port 10052 to zabbix-server container https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/templates/deployment-zabbix-server.yaml#L109
I think new deployment with container java-gateway with respective service need to be created. New variables should be added telling Zabbix Server how to connect to java-gateway.

**Comment by aeciopires** [comment]
Hello @BGmot!

Thanks for your report.

I understood that you would like to add new configurations in the zabbix_server.conf file, right?

You can make this using this helm values:

```bash
# **Zabbix Server** configurations
zabbixserver:
  # -- Enables use of **Zabbix Server**
  enabled: true
  # -- Extra environment variables. A list of additional environment variables. See example: https://github.com/zabbix-community/helm-zabbix/blob/master/charts/zabbix/docs/example/kind/values.yaml
  extraEnv: []
    #- name: ENABLE_TIMESCALEDB
    #  value: "true"
  # -- annotations to add to the deployment
  deploymentAnnotations: {}
  # -- annotations to add to the containers
  containerAnnotations: {}
  # -- additional volumeMounts to the zabbix server container
  extraVolumeMounts: []
  # -- additional containers to start within the zabbix server pod
  extraContainers: []
  # -- additional init containers to start within the zabbix server pod
  extraInitContainers: []
  # -- additional volumes to make available to the zabbix server pod
  extraVolumes: []
  # -- additional specifications to the zabbix server pod
  extraPodSpecs: {}
```

Your need is an advanced configuration and we understand that you must have knowledge to execute.

# References:

- https://hub.docker.com/r/zabbix/zabbix-server-pgsql
- https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/
- https://github.com/criblio/helm-charts/blob/master/common_docs/EXTRA_EXAMPLES.md
- https://github.com/cetic/helm-zabbix/pull/73

**Comment by BGmot** [comment]
Hello @aeciopires !
Java gateway is one of components in Zabbix world (like zabbix-server, zabbix-agent, zabbix-web) so ideally all these should be different services/deployments. Coupling java-gateway with zabbix-server in the same deployment is not a good idea in my opinion. As of now I don't know how I can achieve that with helm values (probably with extraContainers?), I'm new to helm, I'll dig into that, thanks.
In any case current port forward for port 10052 is useless.

**Comment by aeciopires** [comment]
Hello @BGmot!

I understood your need. Looking the Docker Hub, the Zabbix SIA published the image for Zabbix Java Gateway. 
I can do this, but I will need some weeks. You would like learn Helm and open a Pull Request about this?

# References:

- https://hub.docker.com/r/zabbix/zabbix-java-gateway 
- https://helm.sh/docs

**Comment by BGmot** [comment]
> You would like learn Helm and open a Pull Request about this?

Yes.

**Comment by sa-ChristianAnton** [comment]
This is an interesting request, and YES, we should support Java Gateway, ... I have looked around a bit and found there is a bit of legacy laying around which we would have to clean up and some ideas on how to do it...

Legacy laying around is mostly the `ZBX_JAVAGATEWAY_ENABLE` var which is defined two times within `values.yaml` but not being used in any useful way at all:
* the definition in `values.yaml` is in the "zabbixproxy" and the "zabbixagent" sections. Ladder is nonsense, because the Java Gateway is to be deployed within Server or Proxy, but never within an agent.
* in the template for the Zabbix proxy, this variable is being used, but in a wrong way.
* the port 10052 forwarding without any real checking whether it is needed or not, mentioned by @BGmot 

Now, regarding having the Java Gateway within the same pod as the Zabbix server or/and proxy, I actually agree with @BGmot that it would be better to deploy this as an extra deployment. BUT we should then make sure that we do NOT support to deploy both zabbix server AND proxy at the same time with this chart, that would make it easier from the naming perspective for services, i.e.

If we look into the entrypoint.sh of the zabbix server image, for example, we find this:
```yaml
    : ${ZBX_JAVAGATEWAY_ENABLE:="false"}
    if [ "${ZBX_JAVAGATEWAY_ENABLE,,}" == "true" ]; then
        update_config_var $ZBX_CONFIG "JavaGateway" "${ZBX_JAVAGATEWAY:-"zabbix-java-gateway"}"
        update_config_var $ZBX_CONFIG "JavaGatewayPort" "${ZBX_JAVAGATEWAYPORT}"
        update_config_var $ZBX_CONFIG "StartJavaPollers" "${ZBX_STARTJAVAPOLLERS:-"5"}"
    else
        update_config_var $ZBX_CONFIG "JavaGateway"
        update_config_var $ZBX_CONFIG "JavaGatewayPort"
        update_config_var $ZBX_CONFIG "StartJavaPollers"
    fi
```

so, if we enable the `ZBX_JAVAGATEWAY_ENABLE` env var within the server or proxy container, it will configure itself to try to connect to a java gateway under the name "zabbix-java-gateway" which could be just a service name enabled. Then of course it should not happen that both Zabbix server AND proxy are enabled and both try to access the same java gateway deployment. But I believe it is nonsense anyway to deploy server and proxy within the same namespace and from the same helm chart, so that should not be a problem.

I would like to do this implementation, even if I don't know when exactly I will do so.... maybe in a week or so.

**Comment by aeciopires** [comment]
Hello @BGmot and @sa-ChristianAnton!

@mohammed6688 implemented the Zabbix Java Gateway installation in the PR #78 and I fixed some bugs in the PR #82.

Please install helm chart version 4.3.0
